### PR TITLE
Add Hdfs support in DataSink

### PR DESCRIPTION
**Pull request by JkSelf**
When we support the native parquet write function in Gluten, we found that Velox Parquet Writer only support write the data into local file system. And this PR add the  HDFS file system writing support. And then Gluten can write the data into HDFS path when calling velox parquet writer.

### CODE REVIEW

**Comment by JkSelf** [comment]
@zhejiangxiaomai  @Yuhta Please help to review.

**Comment by majetideepak** [comment]
@JkSelf can you rebase this branch with the main branch?

File: velox/connectors/hive/storage_adapters/hdfs/HdfsFileSink.h [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+
+  ~HdfsFileSink() override {
+    destroy();
+  }
+
+  using facebook::velox::dwio::common::DataSink::write;
+
+  void write(std::vector<facebook::velox::dwio::common::DataBuffer<char>>&
+                 buffers) override;
+
```
Comment (majetideepak) [inline velox/connectors/hive/storage_adapters/hdfs/HdfsFileSink.h:51]: Is this defined?

File: velox/connectors/hive/storage_adapters/hdfs/HdfsFileSink.h [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+  static void registerFactory();
```
Comment (majetideepak) [inline velox/connectors/hive/storage_adapters/hdfs/HdfsFileSink.h:52]: Is this defined?

**Comment by JkSelf** [comment]
[majetideepak](https://github.com/majetideepak) Thanks for your review. 
https://github.com/facebookincubator/velox/pull/5663 already implemented the HDFS support. And I will close this PR.

**Pull request closed without merging**